crypto/pkcs11key/key.go

```diff
@@ -198,10 +241,28 @@ func (ps *Key) getPrivateKey(module ctx, session pkcs11.SessionHandle, label str
 	}
 	privateKeyHandle := objs[0]
 
+	attributes, err := module.GetAttributeValue(session, privateKeyHandle, []*pkcs11.Attribute{
+		pkcs11.NewAttribute(pkcs11.CKA_KEY_TYPE, false),
+	})
+	if err != nil {
+		return noHandle, err
+	}
+	for i, attribute := range attributes {
```

Comment:
I tried to stay consistent with the rest of the file. The Block reading CKA_ALWAYS_AUTHENTICATE did it like that.